yaks/lib/yaks/format/json_api.rb

```diff
@@ -38,7 +38,7 @@ def serialize_resource(resource)
       def serialize_links(subresources)
         subresources.each_with_object({}) do |resource, hsh|
           next if resource.null_resource?
-          hsh[resource.type] = serialize_link(resource)
+          hsh[resource.rels.first] = serialize_link(resource)
```

Comment:
So yeah I just read the manual...

> `:rel` : Set the relation (symbol or URI) this association has with the object. Will be derived from the association name and the configured rel_template if ommitted

In the context of JSON-API, I would assume that I would do something like this in the mapper:

```ruby
class CustomerMapper < Yaks::Mapper
# ...
  has_many :books, rel: :favourite_books
# ...
end
```

Loading a customer record and calling `yaks.map` on it, I looked at the subresources, and it had a `rels=[:favourite_books]` on the Yaks::CollectionResource. However if `rel:` is not defined in the mapper then the sub resource will have `rels=[]`, which would mean **I broke the formatter.** 

Any suggestions for where I could start adding more tests? Also should rels not be empty at this stage? I'm assuming that the formatter should not know or care about the mapper.